### DISCUSSION TITLE
如何保留参考文献标题中的空格，且在其页眉中删去空格

### DISCUSSION THREAD

**Discussion by jchWill** [sjtug/SJTUThesis#967] [提问] [answered]
最终预期效果：
![image](https://github.com/sjtug/SJTUThesis/assets/79404073/126a989b-2308-4916-b1bc-05b883d44231)

然而我的做法会有副作用。我的参考文献超过一页，我尝试在

```latex
\begin{center}
\setstretch{1.15}
\printbibliography[heading=bibintoc]
\end{center}
```

之前使用了

```latex
\fancyhead[R]{参考文献}
```

发现其将上一章最后一页的页眉也改成了**参考文献**。
![image](https://github.com/sjtug/SJTUThesis/assets/79404073/1476f1a4-0740-4b40-8d2d-e322ff61eaa7)
之后我尝试在上一章末尾使用

```latex
\clearpage
```

发现“参考文献”内容之前出现了一面带参考文献页眉和页码的空白页，请问如何解决？

**Comment by AlexaraWu** [comment, marked as answer]
```latex
\makeatletter
\defbibheading{custom}[\bibname]{%
  \SJTU@head[参考文献]{#1}}
\makeatother

\printbibliography[heading=custom]
```

**Reply by jchWill** [reply]
这样的话目录里的参考文献四个字会没有空格，请问需要怎么修改呢，或者如何在不影响上述效果的前提下将原来目录项去掉，我自己addcontentsline

**Reply by AlexaraWu** [reply]
给 `\SJTU@head` 再加一个参数：`\SJTU@head[参考文献]{#1}[#1]`。

**Reply by jchWill** [reply]
感谢解答 :smile_cat:

**Reply by frelwx** [reply]
请问是直接替换`main.tex`中的`\printbibliography[heading=bibintoc]`吗，我试了下似乎没有达到为“参考文献”添加空格的效果😭

**Reply by jchWill** [reply]
> 请问是直接替换`main.tex`中的`\printbibliography[heading=bibintoc]`吗，我试了下似乎没有达到为“参考文献”添加空格的效果😭

在`setup.tex`中，替换`\sjtusetup`中的`name`字段为

``` latex
  name = {
    bib = {参\hspace{\ccwd}考\hspace{\ccwd}文\hspace{\ccwd}献}
  },
```
再加上上述方法。
